packages/transaction-controller/src/TransactionController.ts

```diff
@@ -644,21 +674,21 @@ export class TransactionController extends BaseController<
 
       const chainId = parseInt(currentChainId, undefined);
       const { approved: status } = TransactionStatus;
-
-      const txNonce =
-        nonce ||
-        (await query(this.ethQuery, 'getTransactionCount', [from, 'pending']));
+      let nonceToUse = nonce;
+      // if a nonce already exists on the transactionMeta it means this is a speedup or cancel transaction
+      // so we want to reuse that nonce and hope that it beats the previous attempt to chain. Otherwise use a new locked nonce
+      if (!nonceToUse) {
```

Comment:
@Gudahtt I'm realizing that this isn't actually necessary in the current implementation since in this `TransactionController` we don't actually re-process/re-approve speedup and cancel transactions in this method [as we do in the extension `TransactionController`](https://github.com/MetaMask/metamask-extension/blob/6638d799327283f635085c2bfc933d779de602f5/app/scripts/controllers/transactions/index.js#L1303). Rather we just modify the transaction and re-broadcast it directly in the speedup/cancel method, for instance [here](https://github.com/MetaMask/core/blob/d55867aa50704cb22a65f7a9231e66dcc09ddc06/packages/transaction-controller/src/TransactionController.ts#L920).

That said it may make sense to leave this logic to ensure correct nonce management in future uses of this method. And as it is currently there is no issue keeping this logic since `nonceToUse` should always be undefined on line 680.